lib/improver/nbhood.py

```diff
@@ -195,6 +198,115 @@ def __repr__(self):
         return result.format(self.unweighted_mode)
 
     @staticmethod
+    def mean_over_neighbourhood(cube, cells_x, cells_y):
+        """
+        Method to calculate the average value in a square neighbourhood using
+        the 4-point algorithm to find the total sum over the neighbourhood.
+
+        The output from the cumulate_array method can be used to
+        calculate the sum over a neighbourhood of size cells_x*cells_y. This
+        sum is then divided by the area of the neighbourhood to calculate the
+        mean value in the neighbourhood.
+
+        At edge points, the sum and area of the neighbourhood are calculated
+        for each point individually. For non-edge points, a faster, vectorised
+        approach is taken:
+        1. The displacements between the four points used to calculate the
+           neighbourhood total sum and the central grid point are calculated.
+        2. Four copies of the cumulate array output are flattened and rolled
+           by these displacements to align the four terms used in the
+           neighbourhood total sum calculation.
+        3. The neighbourhood total at all non-edge points can then be
+           calculated simultaneously in a single vector sum.
+
+        Parameters
+        ----------
+        cube : iris.cube.Cube
+            Cube to which neighbourhood processing is being applied. Must
+            be passed through cumulate_array method first.
+        cells_x, cells_y : integer
```

Comment:
This needs clarifying whether these are the radius or diameter, I think Nigel specifies these values to be the number of grid points in the radius, not including the central grid point, so values of 2 will give a 5x5 neighbourhood.  Is this the convention you've used?